spec/support/specialist_publisher_assertion_helpers.rb

```diff
@@ -47,11 +47,43 @@ def expect_discarded_draft
     expect(find(".alert").text).to match(/^Discarded/)
   end
 
+  def expect_attachment_removed
+    expect(find(".alert").text).to match(/^Attachment/)
+  end
+
   def expect_published_document
     expect(page).to have_content("There are no changes to publish.")
     expect_view_on_website_link
   end
 
+  def expect_add_attachment
+    within(".page-header")do
+      expect(page).to have_content("Add attachment")
+    end
+  end
+
+  def expect_attached_file_frontend
+    reload_page_while_failing do
```

Comment:
This is a broader suggestion than what your PR contains, but I wonder if we can set up a method that does the reload_page_while_failing to find the specialist-frontend meta tag and use that as a wait mechanism to check frontend is up to date.

Once we did that we could make the checks for content simpler and not need to use selectors.